model/task/task_test.go

```diff
@@ -1253,6 +1253,43 @@ func TestBulkInsert(t *testing.T) {
 	}
 }
 
+func TestGetTaskIdBetweenIds(t *testing.T) {
+	assert := assert.New(t)
+	require.NoError(t, db.ClearCollections(Collection))
+	displayName := "cool-task-9000"
+	buildVarient := "bv"
+	requester := "r"
+	project := "proj"
+	tasks := []Task{}
+	for i := 1; i <= 20; i++ {
+		task := Task{
+			Id:                  "t" + fmt.Sprint(i),
+			DisplayName:         displayName,
+			BuildVariant:        buildVarient,
+			Requester:           requester,
+			Project:             project,
+			RevisionOrderNumber: i,
+		}
+		assert.NoError(task.Insert())
+		tasks = append(tasks, task)
+	}
+	t10, err := GetTaskIdBetweenTasks(tasks[0], tasks[19])
+	assert.Equal(t10.RevisionOrderNumber, 10)
+	assert.NoError(err)
+
+	t5, err := GetTaskIdBetweenTasks(tasks[0], tasks[9])
+	assert.Equal(t5.RevisionOrderNumber, 5)
+	assert.NoError(err)
+
+	t15, err := GetTaskIdBetweenTasks(tasks[10], tasks[19])
+	assert.Equal(t15.RevisionOrderNumber, 15)
+	assert.NoError(err)
+
+	t19, err := GetTaskIdBetweenTasks(tasks[17], tasks[19])
+	assert.Equal(t19.RevisionOrderNumber, 19)
+	assert.NoError(err)
```

Comment:
May be worth testing other valid input cases like:
* Passing a higher order number task as the first arg.
* Two of the same task.
* Tasks that are adjacent in terms of revision number.
* Passing a task that has no lower revision order number/no other matching BV, etc (i.e. it's the first version in the repo).

Reply:
I'm not sure what you mean by the last one, like it's the one and only task ever for that project, etc..?
I think the new test cases for display name and etc handle this, since if a task is the only one for that project, there cannot be another one with the same display name, project, bv, etc

Reply:
Yeah, the last case isn't really possible now that the validation is there, so I don't think it needs to be tested explicitly.